src/executor/values.rs

```diff
@@ -17,17 +19,32 @@ impl ValuesExecutor {
     #[try_stream(boxed, ok = DataChunk, error = ExecutorError)]
     pub async fn execute(self) {
         for chunk in self.values.chunks(PROCESSING_WINDOW_SIZE) {
+            type Type = DataTypeKind;
             // Create array builders.
-            let mut builders = self
-                .column_types
+            let column_types = &self.column_types;
+            let mut builders = column_types
                 .iter()
                 .map(|ty| ArrayBuilderImpl::with_capacity(chunk.len(), ty))
                 .collect_vec();
             // Push value into the builder.
             let dummy = DataChunk::single(0);
             for row in chunk {
-                for (expr, builder) in row.iter().zip_eq(&mut builders) {
+                for (expr, column_type, builder) in izip!(row, column_types, &mut builders) {
                     let value = expr.eval(&dummy)?;
+                    let size = match column_type.kind {
+                        Type::Varchar(size) => size,
+                        Type::Char(size) => size,
+                        _ => None,
+                    };
+                    if let Some(width) = size {
+                        let item_length = value.get(0).len() as u64;
```

Comment:
If `width` is `Some`, we are sure that the `value` will be a `String`. Therefore, we can:

let item_length = if let DataValue::String(x) = value.get(0) { x.len() as u64 } else if let DataValue::Null = value.get(0) { 0 } else { unreachable!() };